python/spider-py/src/spider_py/core/task.py

```diff
@@ -1,5 +1,50 @@
 """Task module for Spider."""
 
+from dataclasses import dataclass, field
+from enum import IntEnum
+from uuid import UUID, uuid4
 
+from spider_py.core.data import DataId
+
+TaskId = UUID
+
+
+@dataclass
+class TaskInputOutput:
+    """Represents a task input that points to output of another task"""
```

Comment:
As discussed offline, this class name is a bit confusing. But since we're mirroring C++ names, let's keep it untouched for now. For the docstring, how about:
```suggestion
    """Represents a task input that references the output of another task by its ID and position."""
```
This is revised based on my understanding. If it's not precise, feel free to iterate.